### PR TITLE
Start Testing

### PR DESCRIPTION
**Pull request by maxxbay**
Learning testing in weekend

### CODE REVIEW

**Comment by lenaricabal** [comment]
image

File: src/components/GenreSelect.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import '@testing-library/jest-dom/extend-expect';
+import GenreSelect from './GenreSelect';
+
+describe('GenreSelect', () => {
+  const genres = ['Action', 'Comedy', 'Drama'];
+
+  test('renders all genres passed in props', () => {
+    render(<GenreSelect genres={genres} />);
+    genres.forEach(genre => {
+      expect(screen.getByText(genre)).toBeInTheDocument();
```
Comment (lenaricabal) [inline src/components/GenreSelect.test.js:12]: when you get element by text it's better to use TextMatch or add {exact:false}

File: src/components/GenreSelect.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    });
+  });
+
+  test('highlights a selected genre passed in props', () => {
+    render(<GenreSelect genres={genres} selectedGenre="Drama" />);
+    const selectElement = screen.getByTitle('Genre');
+    expect(selectElement.value).toBe('Drama');
+  });
+
+  test('after a click event on a genre button component calls "onSelect" callback and passes correct genre in arguments', () => {
```
Comment (lenaricabal) [inline src/components/GenreSelect.test.js:22]: please, make description shorter

File: src/components/SearchForm.test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import SearchForm from './SearchForm';
+
+describe('SearchForm', () => {
+  test('renders an input with the value equal to initial value passed in props', () => {
+    render(<SearchForm initialSearchQuery="Initial search" />);
+    const input = screen.getByPlaceholderText('What do you want to watch?');
```
Comment (lenaricabal) [inline src/components/SearchForm.test.js:9]: in this case it's better to use getByRole

File: src/components/SearchForm.test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    expect(input.value).toBe('Initial search');
+  });
+
+  test('after typing to the input and a "click" event on the Submit button, the "onSearch" prop is called with proper value', () => {
```
Comment (lenaricabal) [inline src/components/SearchForm.test.js:13]: please, make description shorter

File: src/components/SearchForm.test.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    const input = screen.getByPlaceholderText('What do you want to watch?');
+    const submitButton = screen.getByText('search');
+
+    fireEvent.change(input, { target: { value: 'New search' } });
+    fireEvent.click(submitButton);
+
+    expect(onSearchMock).toHaveBeenCalledWith('New search');
+  });
+
+  test('after typing to the input and pressing Enter key, the "onSearch" prop is called with proper value', () => {
```
Comment (lenaricabal) [inline src/components/SearchForm.test.js:27]: same here

File: src/components/SearchForm.test.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    const onSearchMock = jest.fn();
+    render(
+      <SearchForm initialSearchQuery="Initial search" onSearch={onSearchMock} />
+    );
+    const input = screen.getByPlaceholderText('What do you want to watch?');
+
+    fireEvent.change(input, { target: { value: 'New search' } });
+    fireEvent.submit(input);
```
Comment (lenaricabal) [inline src/components/SearchForm.test.js:35]: please, use userEvent to check `enter` keydown

File: src/components/SearchForm.test.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    expect(input.value).toBe('Initial search');
+  });
+
+  test('after typing to the input and a "click" event on the Submit button, the "onSearch" prop is called with proper value', () => {
+    const onSearchMock = jest.fn();
+    render(
+      <SearchForm initialSearchQuery="Initial search" onSearch={onSearchMock} />
+    );
+    const input = screen.getByPlaceholderText('What do you want to watch?');
+    const submitButton = screen.getByText('search');
```
Comment (lenaricabal) [inline src/components/SearchForm.test.js:19]: if you search by text better use TextMatch

**Comment by lenaricabal** [comment]
your `build` and `coverage` is broken, it's good to fix it

File: src/components/GenreSelect.test.js (hunk 1 above)
Comment (maxxbay) [reply src/components/GenreSelect.test.js:12]: done

File: src/components/GenreSelect.test.js (hunk 2 above)
Comment (maxxbay) [reply src/components/GenreSelect.test.js:22]: ok

File: src/components/SearchForm.test.js (hunk 3 above)
Comment (maxxbay) [reply src/components/SearchForm.test.js:9]: done

File: src/components/SearchForm.test.js (hunk 4 above)
Comment (maxxbay) [reply src/components/SearchForm.test.js:13]: +

File: src/components/SearchForm.test.js (hunk 5 above)
Comment (maxxbay) [reply src/components/SearchForm.test.js:27]: +

File: src/components/SearchForm.test.js (hunk 6 above)
Comment (maxxbay) [reply src/components/SearchForm.test.js:35]: done

**Pull request merged**